lib/Dialect/RTL/RTLOps.cpp

```diff
@@ -775,7 +775,8 @@ void InstanceOp::getAsmResultNames(OpAsmSetValueNameFn setNameFn) {
   size_t baseNameLen = name.size();
 
   for (size_t i = 0, e = getNumResults(); i != e; ++i) {
-    auto resName = getModuleResultName(module, i);
+    StringRef resName =
+        module->hasAttr("resultNames") ? getModuleResultName(module, i) : "";
```

Comment:
@lattner Is this kind of defensiveness okay? The dialect conversion framework has this `--debug-only=dialect-conversion` feature which is pretty handy to observe/debug the conversion. However, it uses the custom assembly printer to show the assembly in between steps, where the IR might be in an inconsistent state. Without this defensive check, debugging the conversion will crash if it is translating modules from one op to another. What is the idiomatic fix here? I see as options:
1. Make printers more defensive (as in this change)
2. Use the generic assembly format in dialect conversion code upstream
3. Make asm printers independent of referenced modules by replicating some of the information (that seems closer to what MLIR itself does).

Reply:
The problem with that approach is that it leads to a lot of defensive code spread throughout the compiler, and it is very difficult (and not worth it) to test.  MLIR has a `--mlir-print-op-generic` which disables the custom printers, I think that is a better and more general way to handle this sort of debugging scenario.

That said, I see this flag on mlir-opt, but I don't see it getting plumbed through firtool/circt-opt.  It looks like we're missing some registration call.

Reply:
> That said, I see this flag on mlir-opt, but I don't see it getting plumbed through firtool/circt-opt.

Ah that's the reason I didn't see it. Perfect, that fixes the issue! Thanks for pointing it out. I'll take a look where we're missing the registration calls.